recipes/libcurl/all/test_package/conanfile.py

```diff
@@ -1,30 +1,36 @@
-from conans import ConanFile, CMake, tools
 import os
 import subprocess
 import re
 
+from conans import ConanFile, tools
+from conan.tools.cmake import CMake
+from conan.tools.layout import cmake_layout
+
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "cmake_find_package"
+    generators = "CMakeDeps", "CMakeToolchain"
 
     def build(self):
         if tools.cross_building(self.settings) and self.settings.os in ["iOS"]:
-            return  # on iOS I do not even need to build, it will just give am a and error about unsigned binaries       
+            return  # on iOS I do not even need to build, it will just give am a and error about unsigned binaries
         cmake = CMake(self)
         cmake.configure()
         cmake.build()
 
+    def layout(self):
+        cmake_layout(self)
+
     @property
     def _test_executable(self):
         if self.settings.os == "Windows":
-            return os.path.join("bin", "test_package.exe")
+            return os.path.join(self.cpp.build.bindirs[0], "test_package.exe")
```

Comment:
If we're not modifying `self.cpp.build.bindirs`, isn't the default always `bin`?
I don't want to always write `self.cpp.build.bindirs[0]`.

Reply:
The thing is, you are not modifying `bindirs` but the `cmake_layout(self)` does. Now cmake is not adjusting any output directory (we want to keep the cmake build more similar to the original) so with multi-configuration build systems is using a "Release/Debug" folder and nothing otherwise.

```
def layout(self):
        cmake_layout(self)
```